File: src/main/java/net/hofmann/example/springtriangle/data/NumberCalculatorImpl.java
package net.hofmann.example.springtriangle.data;

import org.springframework.context.annotation.Primary;
import org.springframework.stereotype.Component;

import java.util.Collections;
import java.util.List;

@Component
public class NumberCalculatorImpl implements NumberCalculator {

    final static List<Integer> tenNumbers = Collections.unmodifiableList(
            List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));

    @Override
    public List<Integer> numbers() {
        return tenNumbers;
    }
}

File: src/main/java/net/hofmann/example/springtriangle/data/NumberCalculatorImpl.kt
package net.hofmann.example.springtriangle.data

import org.springframework.stereotype.Component
import java.util.*

@Component
class NumberCalculatorImpl : NumberCalculator {
    override fun numbers(): List<Int> {
        return tenNumbers
    }

    companion object {
        val tenNumbers = Collections.unmodifiableList(
                java.util.List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
    }
}